packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -52,84 +51,74 @@ const InsertFromURLPopover = ( { src, onChange, onSubmit, onClose } ) => (
 	</URLPopover>
 );
 
-export class MediaPlaceholder extends Component {
-	constructor() {
-		super( ...arguments );
-		this.state = {
-			src: '',
-			isURLInputVisible: false,
-		};
-		this.onChangeSrc = this.onChangeSrc.bind( this );
-		this.onSubmitSrc = this.onSubmitSrc.bind( this );
-		this.onUpload = this.onUpload.bind( this );
-		this.onFilesUpload = this.onFilesUpload.bind( this );
-		this.openURLInput = this.openURLInput.bind( this );
-		this.closeURLInput = this.closeURLInput.bind( this );
-	}
-
-	onlyAllowsImages() {
-		const { allowedTypes } = this.props;
-		if ( ! allowedTypes ) {
-			return false;
-		}
-		return every( allowedTypes, ( allowedType ) => {
-			return (
-				allowedType === 'image' || startsWith( allowedType, 'image/' )
-			);
-		} );
-	}
-
-	componentDidMount() {
-		this.setState( { src: get( this.props.value, [ 'src' ], '' ) } );
-	}
+export function MediaPlaceholder( {
+	value = {},
+	allowedTypes = [],
+	className,
+	icon,
+	labels = {},
+	mediaPreview,
+	notices,
+	isAppender,
+	accept,
+	addToGallery,
+	multiple = false,
+	dropZoneUIOnly,
+	disableDropZone,
+	disableMediaButtons,
+	onError,
+	onSelect,
+	onCancel,
+	onSelectURL,
+	onDoubleClick,
+	onFilesPreUpload = noop,
+	onHTMLDrop = noop,
+	children,
+} ) {
+	const mediaUpload = useSelect( ( select ) => {
+		const { getSettings } = select( 'core/block-editor' );
+		return getSettings().mediaUpload;
+	}, [] );
+	const [ src, setSrc ] = useState( '' );
+	const [ isURLInputVisible, setIsURLInputVisible ] = useState( false );
+
+	useEffect( () => {
```

Comment:
Try using `useLayoutEffect` here, which will change the `src` value synchronously before rendering, unlike `useEffect` which changes the value in the next render. `useLayoutEffect` will apply the change earlier, which is closer to the behavior of `componentDidMount` and `componentDidUpdate`. I suspect that updating this to use `useLayoutEffect` will get tests to pass, judging by @ItsJonQ's comment about the tests passing if they're run in slow-mode.

Reply:
This is good advice. :)

Reply:
Quoting [React doc](https://reactjs.org/docs/react-component.html#componentdidmount):

> You may call `setState()` immediately in `componentDidMount()`. It will trigger an extra rendering, but it will happen before the browser updates the screen. This guarantees that even though the `render()` will be called twice in this case, the user won’t see the intermediate state.

Reply:
I had already tried that, but still fails

Reply:
It seems to be related with puppeteer and `useEffect`, but `slowmo` isn't required for the test to pass. Just `interactive mode`. I'm struggling with it all day now and haven't pinpointed yet.

It also succeeds if I introduce a delay between  insertBlock and press Backspace. But not a small one or `0`, so it doesn’t have to do with just the next tick  — example to pass await new Promise( ( res ) => setTimeout( res, 200 ) );

Reply:
In our case though it shouldn't matter because `src` is not relevant to the empty placeholder. That explains why the test still fails. It has to do with the `backspace` I believe, but doesn't make any sense why it passes in master.